packages/dds/tree/src/events/events.ts

```diff
@@ -153,22 +154,22 @@ export class EventEmitter<E extends Events<E>> implements IEventEmitter<E> {
      * @returns a function which will deregister the listener when run
      */
     public on<K extends keyof Events<E>>(eventName: K, listener: E[K]): () => void {
-        const listeners = this.listeners[eventName];
+        const listeners = this.listeners.get(eventName);
         if (listeners !== undefined) {
             listeners.add(listener);
         } else {
-            this.listeners[eventName] = new Set([listener]);
+            this.listeners.set(eventName, new Set([listener]));
         }
 
         return this.off.bind(this, eventName, listener);
     }
 
     private off<K extends keyof Events<E>>(eventName: K, listener: E[K]): void {
-        const listeners = this.listeners[eventName];
+        const listeners = this.listeners.get(eventName);
         if (listeners !== undefined) {
             listeners.delete(listener);
```

Comment:
I think this should assert if the item is not subscribed instead of ignoring it. Something like:

```suggestion
        const listeners = this.listeners.get(eventName) ?? fail("not subscribers for event";
        const deleated = listeners.delete(listener);
        assert(deleated, "subscriber not found");
```

Reply:
That means that an unsubscribe function cannot be called more than once. Should that be part of the contract of the `IEventEmitter` interface, or an implementation detail of `EventEmitter` (and the behavior for `IEventEmitter` is documented as undefined)?

Reply:
I went with the latter approach.